src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -427,5 +529,178 @@ private ImmutableArray<BoundExpression> BindInterpolatedStringParts(BoundUnconve
+
+            if (memberAnalysisResult.ArgsToParamsOpt.IsDefault && arguments.Count == parameters.Length)
+            {
+                // No parameters are missing and no remapped indexes, we can just use the original indexes
+                handlerArgumentIndexes = handlerParameterIndexes;
+            }
+            else
+            {
+                // Args and parameters were reordered via named parameters, or parameters are missing. Find the correct argument index for each parameter.
+                var handlerArgumentIndexesBuilder = ArrayBuilder<int>.GetInstance(handlerParameterIndexes.Length, fillWithValue: BoundInterpolatedStringArgumentPlaceholder.UnspecifiedParameter);
+                for (int argumentIndex = 0; argumentIndex < arguments.Count; argumentIndex++)
+                {
+                    // The index in the original parameter list we're looking to match up.
+                    var argumentParameterIndex = memberAnalysisResult.ParameterFromArgument(argumentIndex);
+                    for (int handlerParameterIndex = 0; handlerParameterIndex < handlerParameterIndexes.Length; handlerParameterIndex++)
+                    {
+                        // Is the original parameter index of the current argument the parameter index that was specified in the attribute?
+                        if (argumentParameterIndex == handlerParameterIndexes[handlerParameterIndex])
+                        {
+                            // We can't just bail out on the first match: users can duplicate parameters in attributes, causing the same value to be passed twice.
+                            Debug.Assert(handlerArgumentIndexesBuilder[handlerParameterIndex] == BoundInterpolatedStringArgumentPlaceholder.UnspecifiedParameter);
+                            handlerArgumentIndexesBuilder[handlerParameterIndex] = argumentIndex;
+                        }
+                    }
+                }
+
+                handlerArgumentIndexes = handlerArgumentIndexesBuilder.ToImmutableAndFree();
+            }
+
+            var argumentPlaceholdersBuilder = ArrayBuilder<BoundInterpolatedStringArgumentPlaceholder>.GetInstance(handlerArgumentIndexes.Length);
+            var argumentRefKindsBuilder = ArrayBuilder<RefKind>.GetInstance(handlerArgumentIndexes.Length);
+            bool hasErrors = false;
+
+            // Now, go through all the specified arguments and see if any were specified _after_ the interpolated string, and construct
+            // a set of placeholders for overload resolution.
+            for (int i = 0; i < handlerArgumentIndexes.Length; i++)
+            {
+                int argumentIndex = handlerArgumentIndexes[i];
+                Debug.Assert(argumentIndex != interpolatedStringArgNum);
+
+                RefKind refKind;
+                TypeSymbol placeholderType;
+                if (argumentIndex == BoundInterpolatedStringArgumentPlaceholder.InstanceParameter)
```

Comment:
>BoundInterpolatedStringArgumentPlaceholder.InstanceParameter [](http://example.com/codeflow?start=37&length=60)

It is quite possible I am missing something, but where do we put this value into the ```handlerArgumentIndexes``` for "Args and parameters were reordered via named parameters, or parameters are missing" scenario?

Reply:
You're not missing anything, I am though.

Reply:
Added a test to cover this.